Explorer/Assets/DCL/Settings/ModuleControllers/MemoryLimitSettingController.cs

```diff
@@ -18,12 +18,20 @@ public MemoryLimitSettingController(SettingsDropdownModuleView view, ISystemMemo
             this.view = view;
             this.systemMemoryCap = systemMemoryCap;
 
-            view.DropdownView.Dropdown.value = settingsDataStore.HasKey(MEMORY_CAP_DATA_STORE_KEY)
-                ? settingsDataStore.GetDropdownValue(MEMORY_CAP_DATA_STORE_KEY)
-                : GetIndexFromMemoryCap(systemMemoryCap.MemoryCapInMB);
-
+            if (settingsDataStore.HasKey(MEMORY_CAP_DATA_STORE_KEY))
+            {
+                int value = settingsDataStore.GetDropdownValue(MEMORY_CAP_DATA_STORE_KEY);
+                if (value < view.DropdownView.Dropdown.options.Count)
+                    view.DropdownView.Dropdown.value = value;
+                else
+                    GetIndexFromMemoryCap(systemMemoryCap.MemoryCapInMB);
+            }
+            else
+            {
+                GetIndexFromMemoryCap(systemMemoryCap.MemoryCapInMB);
+            }
```

Comment:
I think you can merge 2 if together by inlining `value`

Reply:
also, you missing an assignment 😅

Reply:
Done!